tripy/examples/resnet50/resnet50.ipynb

```diff
@@ -0,0 +1,483 @@
+    "            lambda x: tp.flatten(x, start_dim=1),\n",
+    "            tp.Linear(2048, num_classes),\n",
+    "        )\n",
+    "\n",
+    "    def __call__(self, x: tp.Tensor) -> tp.Tensor:\n",
+    "        features = self.resnet(x)\n",
+    "        return self.classifier(features)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Instantiate the Tripy model"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 16,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "tripy_model = ResNetClassifier()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Load Weights to Tripy Model"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from transformers import ResNetForImageClassification\n",
+    "\n",
+    "# Load the pretrained ResNet50 model from Hugging Face\n",
+    "resnet_pretrained = ResNetForImageClassification.from_pretrained(\n",
+    "    \"microsoft/resnet-50\"\n",
+    ").to(\"cuda\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def load_pretrained_weights(model, pretrained_state_dict):\n",
+    "    model_state_dict_keys = set(model.state_dict().keys())\n",
+    "    model_dict = {}\n",
+    "\n",
+    "    # Convert each weight to a tp.Parameter and update the model's state dict\n",
+    "    for k, v in pretrained_state_dict.items():\n",
+    "        if \"num_batches_tracked\" in k:\n",
+    "            continue  # Skip num_batches_tracked since it is not needed for\n",
+    "            # inference (tp.BatchNorm does not support)\n",
+    "\n",
+    "        assert k in model_state_dict_keys\n",
+    "        model_dict[k] = tp.Parameter(v.contiguous())\n",
+    "\n",
+    "    model.load_state_dict(model_dict)\n",
+    "    return model\n",
+    "\n",
+    "\n",
+    "# Load pretrained weights to tripy model\n",
+    "tripy_model = load_pretrained_weights(\n",
+    "    tripy_model,\n",
+    "    pretrained_state_dict=resnet_pretrained.state_dict(),\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Compile Tripy Model"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def resnet_classifier(input):\n",
+    "    \"\"\"Inference function to compile. Runs forward path\n",
+    "    and returns argmax of output logits.\"\"\"\n",
+    "    logits = tripy_model(input)\n",
+    "    return tp.argmax(logits)\n",
+    "\n",
+    "\n",
+    "input_shape = [1, 3, 224, 224]\n",
+    "compiled_resnet_classifier = tp.compile(\n",
+    "    resnet_classifier,\n",
+    "    args=[tp.InputInfo(input_shape, dtype=tp.float32)],\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Tripy Demo\n",
```

Comment:
We can drop "Tripy` from all the section titles. e.g. this section could be "Running Inference".

Reply:
dropped